backend/mbql/src/metabase/mbql/schema.clj

```diff
@@ -295,9 +295,12 @@
    :else
    Field))
 
-(def ^:private arithmetic-expressions #{:+ :- :/ :* :coalesce})
+(def ^:private arithmetic-expressions #{:+ :- :/ :* :coalesce :length :round :ceil :floor :abs :power :sqrt :log :expt})
```

Comment:
Why was `:length` reclassified as an arithmetic expression?

Reply:
Based on it's return type so you can use it in arithmetic expressions (which you might want for index calculation in `substring`)

Reply:
This is slightly odd, since functions are usually grouped by their _inputs_. But this is probably OK, since it's not user facing.

Reply:
The whole Schema-as-a-typechecker for MBQL is a bit of a hack and would benefit from a rewrite but that's outside the scope of this PR.